app/src/main/java/org/thoughtcrime/securesms/groups/OpenGroupManager.kt

```diff
@@ -41,11 +41,13 @@ object OpenGroupManager {
         isPolling = true
         val storage = MessagingModuleConfiguration.shared.storage
         val servers = storage.getAllOpenGroups().values.map { it.server }.toSet()
```

Comment:
useless `.toSet()`

Reply:
So the conversion to `Set<T>` is in order to remove duplicates from the collection - `getAllOpenGroups()` returns every community you are a member of, so if you are in multiple rooms on the same server (eg. `Session`, `Oxen`, `Session Updates` all on the `getsession.org` server) the result would have 3 entries for `getsession.org`

The `poll` request actually makes a single `/batch` API call to the server which includes a call for each room on that server (so without the `toSet()` the `forEach` loop would end up redundantly initialising and triggering a `poll` for each copy of the server)

Reply:
ohhh... cool.